packages/admin/admin/src/common/toolbar/actions/ToolbarActionButton.tsx

```diff
@@ -0,0 +1,99 @@
+import { Button, ButtonProps, ComponentsOverrides, IconButton, Tooltip } from "@mui/material";
+import { css, Theme, useTheme, useThemeProps } from "@mui/material/styles";
+import React from "react";
+
+import { createComponentSlot } from "../../../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../../../helpers/ThemedComponentBaseProps";
+import { useWindowSize } from "../../../helpers/useWindowSize";
+
+export type ToolbarActionButtonClassKey = "root" | "tooltip" | "button" | "iconButton" | "text" | "outlined" | "contained";
+
+type ToolbarActionButtonProps = ButtonProps &
+    ThemedComponentBaseProps<{
+        tooltip: typeof Tooltip;
+        iconButton: typeof IconButton;
+        button: typeof Button;
+    }>;
+
+type OwnerState = {
+    variant: "text" | "outlined" | "contained";
+};
+
+const StyledTooltip = createComponentSlot(Tooltip)<ToolbarActionButtonClassKey>({
+    componentName: "ToolbarActionButton",
+    slotName: "tooltip",
+})();
+
+const StyledIconButton = createComponentSlot(IconButton)<ToolbarActionButtonClassKey, OwnerState>({
+    componentName: "ToolbarActionButton",
+    slotName: "iconButton",
+    classesResolver(ownerState) {
+        return [ownerState.variant];
+    },
+})(
+    ({ theme, ownerState }) => css`
+        ${ownerState.variant === "contained" &&
+        css`
+            background: ${theme.palette.primary.main};
+            color: ${theme.palette.primary.contrastText};
+            border-radius: 4px;
+
+            &:hover {
+                background: ${theme.palette.primary.dark};
+            }
+        `}
+
+        ${ownerState.variant === "outlined" &&
+        css`
+            border-radius: 4px;
+            border-width: 1px;
+            border-style: solid;
+            border-color: ${theme.palette.grey[200]};
+
+            &:hover {
+                background-color: ${theme.palette.grey[50]};
+                border-color: ${theme.palette.grey[200]};
+            }
+        `}
+    `,
+);
+
+const StyledButton = createComponentSlot(Button)<ToolbarActionButtonClassKey>({
+    componentName: "ToolbarActionButton",
+    slotName: "button",
```

Comment:
Should this also include the `variant` in the `classesResolver`? 
Then the "normal" button also get's the variants as a class-name, not just the icon-button. 


```suggestion
    slotName: "button",
    classesResolver(ownerState) {
        return [ownerState.variant];
    },
```

Reply:
Done b767ede0870422515dd48686a37983a9788aa476